types/yup/index.d.ts

```diff
@@ -146,16 +158,27 @@ export interface BooleanSchemaConstructor {
 }
 
 // tslint:disable-next-line:no-empty-interface
-export interface BooleanSchema extends Schema<boolean> {}
+export interface BooleanSchema<T = boolean> extends Schema<T> {
+    nullable(isNullable?: true): BooleanSchema<T | null>;
+    nullable(isNullable: false): BooleanSchema<Exclude<T, null>>;
+}
 
 export interface DateSchemaConstructor {
     (): DateSchema;
     new (): DateSchema;
 }
 
-export interface DateSchema extends Schema<Date> {
-    min(limit: Date | string | Ref, message?: TestOptionsMessage): DateSchema;
-    max(limit: Date | string | Ref, message?: TestOptionsMessage): DateSchema;
+export interface DateSchema<T = Date> extends Schema<T> {
```

Comment:
Same here, `T extends Date`. This one matters less because users are _unlikely_ to have their own class derived from `Date` but it happens sometimes.